pkg/capr/planner/manifests.go

```diff
@@ -36,6 +37,15 @@ func (p *Planner) getControlPlaneManifests(controlPlane *rkev1.RKEControlPlane,
 	}
 	result = append(result, clusterAgent)
 
+	mgmtCluster, err := p.managementClusters.Get(controlPlane.Spec.ManagementClusterName)
+	if err != nil {
+		return nil, err
+	}
+	// if we're bootstrapping - just return the cluster-agent manifest
```

Comment:
The logic for retrieving information from the mgmt cluster should not be leaked into the planner -- you can implement an `InfoFunction` that given a controlplane object, can validate that the cluster should not provide controlplane manifests.